File: dotnetweb-dest/MyService/DevService.cs
using System.Collections.Generic;
using System;
using UniRpc.WebApplication;
namespace MyService
{
    public abstract class DevService<Dog>: WebSocketServiceBase
    {
        public DevService()
        {
            __reflection = "MyService.DevService";
            __genericArguments = new List<string>() { typeof(Dog).FullName }.AsReadOnly();
        }
        public abstract bool Send(MyService.Message message);
        public abstract bool Get();
        public abstract float StoryCount();
        public abstract string Test();
        public abstract float AB(double my);
        public abstract System.Collections.Generic.List<long> Gap(int value, bool enabled);
        public abstract byte[] Release<Cat>(Cat cat, Dog dog);
        public abstract MyService.Message GetMessage();
        public abstract void IncomingCall();
        public override BaseMessage __invoke(BaseMessage message)
        {
            switch (message.Method)
            {
                case "Send":
                    {
                        MyService.Message ____message = message.Payload.GetProperty<MyService.Message>("message");
                        return message.ReturnMessage(Send(____message));
                    }
                case "Get":
                    {
                        return message.ReturnMessage(Get());
                    }
                case "StoryCount":
                    {
                        return message.ReturnMessage(StoryCount());
                    }
                case "Test":
                    {
                        return message.ReturnMessage(Test());
                    }
                case "AB":
                    {
                        double ____my = message.Payload.GetProperty<double>("my");
                        return message.ReturnMessage(AB(____my));
                    }
                case "Gap":
                    {
                        int ____value = message.Payload.GetProperty<int>("value");
                        bool ____enabled = message.Payload.GetProperty<bool>("enabled");
                        return message.ReturnMessage(Gap(____value, ____enabled));
                    }
                case "Release":
                    {
                        object ____cat = message.Payload.GetPropertyByReflection("cat");
                        Dog ____dog = message.Payload.GetProperty<Dog>("dog");
                        return message.ReturnMessage(Release(____cat, ____dog));
                    }
                case "GetMessage":
                    {
                        return message.ReturnMessage(GetMessage());
                    }
                case "IncomingCall":
                    {
                        IncomingCall();
                        break;
                    }
            }
            throw new NotImplementedException($"{message.Service}.{message.Method} is not implemented.");
        }
    }
}

File: wsui/src/app/MyService/DevService.ts
import {WebsocketService as $UniRpc_WebsocketService} from "../UniRpc/WebsocketService";
import {WebsocketServiceBase as $UniRpc_WebsocketServiceBase} from "../UniRpc/WebsocketServiceBase";
import {Message as $MyService_Message} from "./Message";
import {Injectable} from "@angular/core";
import {Observable} from "rxjs";
import {map} from "rxjs/operators";
@Injectable({
    providedIn: 'root'
})
export class DevService<Dog> extends $UniRpc_WebsocketServiceBase
{
    public constructor (private __websocketService: $UniRpc_WebsocketService)
    {
        super();
        this.__reflection = "MyService.DevService";
        this.__genericArguments = [];
    }
    public Send(message: $MyService_Message): Observable<boolean> {
        return this.__websocketService.send({
            Service: 'MyService.DevService',
            Method: 'Send',
            GenericArguments: [],
            Payload: {
                message: message
            }
        }).pipe(map(__result => {
            return __result.Payload as boolean;
        }));
    }
    public Get(): Observable<boolean> {
        return this.__websocketService.send({
            Service: 'MyService.DevService',
            Method: 'Get',
            GenericArguments: [],
            Payload: {
            }
        }).pipe(map(__result => {
            return __result.Payload as boolean;
        }));
    }
    public StoryCount(): Observable<number> {
        return this.__websocketService.send({
            Service: 'MyService.DevService',
            Method: 'StoryCount',
            GenericArguments: [],
            Payload: {
            }
        }).pipe(map(__result => {
            return __result.Payload as number;
        }));
    }
    public Test(): Observable<string> {
        return this.__websocketService.send({
            Service: 'MyService.DevService',
            Method: 'Test',
            GenericArguments: [],
            Payload: {
            }
        }).pipe(map(__result => {
            return __result.Payload as string;
        }));
    }
    public AB(my: number): Observable<number> {
        return this.__websocketService.send({
            Service: 'MyService.DevService',
            Method: 'AB',
            GenericArguments: [],
            Payload: {
                my: my
            }
        }).pipe(map(__result => {
            return __result.Payload as number;
        }));
    }
    public Gap(value: number, enabled: boolean): Observable<number[]> {
        return this.__websocketService.send({
            Service: 'MyService.DevService',
            Method: 'Gap',
            GenericArguments: [],
            Payload: {
                value: value,
                enabled: enabled
            }
        }).pipe(map(__result => {
            return __result.Payload as number[];
        }));
    }
    /** release resource */
    public Release<Cat>(/** the cat instance */ cat: Cat, dog: Dog): Observable<string> {
        return this.__websocketService.send({
            Service: 'MyService.DevService',
            Method: 'Release',
            GenericArguments: [],
            Payload: {
                cat: cat,
                dog: dog
            }
        }).pipe(map(__result => {
            return __result.Payload as string;
        }));
    }
    public GetMessage(): Observable<$MyService_Message> {
        return this.__websocketService.send({
            Service: 'MyService.DevService',
            Method: 'GetMessage',
            GenericArguments: [],
            Payload: {
            }
        }).pipe(map(__result => {
            return __result.Payload as $MyService_Message;
        }));
    }
    public IncomingCall(): Observable<void> {
        return this.__websocketService.send({
            Service: 'MyService.DevService',
            Method: 'IncomingCall',
            GenericArguments: [],
            Payload: {
            }
        }).pipe(map(__result => {
            return __result.Payload as void;
        }));
    }
}